aws/resource_aws_appsync_datasource.go

```diff
@@ -194,18 +194,16 @@ func resourceAwsAppsyncDatasourceUpdate(d *schema.ResourceData, meta interface{}
 	input := &appsync.UpdateDataSourceInput{
 		ApiId: aws.String(d.Get("api_id").(string)),
 		Name:  aws.String(d.Get("name").(string)),
+		Type:  aws.String(d.Get("type").(string)),
+	}
+
+	if v, ok := d.GetOk("service_role_arn"); ok {
+		input.ServiceRoleArn = aws.String(v.(string))
 	}
 
 	if d.HasChange("description") {
 		input.Description = aws.String(d.Get("description").(string))
 	}
-	if d.HasChange("service_role_arn") {
```

Comment:
Given this type of change, I'm worried that all parameters must be provided during the update call. We should probably add an acceptance test that has something optional like `description` set, then attempts to update something else like `dynamodb_config`, then verifies that `description` is still the original value as expected.

Reply:
@bflad one fix raises other issues, Could you guide me the way how to handle this?
Since all parameters need to be provided in update call, I am thinking to use Get funcs.